lightly/data/collate.py

```diff
@@ -1010,6 +1011,112 @@ def __init__(
+        solarize_probs:
+            Probability of solarization for each crop category.
+        hf_prob:
+            Probability that horizontal flip is applied.
+        cj_prob:
+            Probability that color jitter is applied.
+        cj_strength:
+            Strength of the color jitter.
+        random_gray_scale:
+            Probability of conversion to grayscale.
+
+    """
+
+    def __init__(
+        self,
+        transforms: List[torchvision.transforms.Compose] = [None, None],
+        crop_sizes: List[int] = [224, 96],
+        crop_min_scales: List[float] = [0.2, 0.05],
+        crop_max_scales: List[float] = [1.0, 0.2],
+        gaussian_blur_probs: List[float] = [0.5, 0.1],
+        gaussian_blur_kernel_sizes: List[float] = [0.1, 0.1],
+        solarize_probs: List[float] = [0.0, 0.2],
+        hf_prob: float = 0.5,
+        cj_prob: float = 1.0,
+        cj_strength: float = 0.5,
+        random_gray_scale: float = 0.2,
+        
+    ):
+        super().__init__()
+        self.gridcreator0 = ReturnGrid(crop_size=crop_sizes[0], crop_min_scale=crop_min_scales[0], crop_max_scale=crop_max_scales[0], hf_prob=hf_prob, N=7)
+        self.gridcreator1 = ReturnGrid(crop_size=crop_sizes[1], crop_min_scale=crop_min_scales[1], crop_max_scale=crop_max_scales[1], hf_prob=hf_prob, N=3)
+
+        color_jitter = T.ColorJitter(
+            0.8 * cj_strength,
+            0.8 * cj_strength,
+            0.4 * cj_strength,
+            0.2 * cj_strength,
+        )
+
+        transforms[0] = T.Compose([
+            T.RandomApply([color_jitter], p=cj_prob),
+            T.RandomGrayscale(p=random_gray_scale),
+            GaussianBlur(prob=gaussian_blur_probs[0], kernel_size=gaussian_blur_kernel_sizes[0]), 
+            RandomSolarization(prob=solarize_probs[0]),
+        ])
+
+        transforms[1] = T.Compose([
+            T.RandomApply([color_jitter], p=cj_prob),
+            T.RandomGrayscale(p=random_gray_scale),
+            GaussianBlur(prob=gaussian_blur_probs[1], kernel_size=gaussian_blur_kernel_sizes[1]), 
+            RandomSolarization(prob=solarize_probs[1]),
+        ])
+
+        self.transforms = transforms
+        
+        
+        
+    def forward(self, batch: List[Tuple[Image.Image, int, str]]) -> Tuple[Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor], List[int], List[str]]:
+
+        """
+        Applies transforms to images in the input batch.
+
+        Args:
+            batch:
+                A list of tuples containing an image (as a PIL Image), a label (int), and a filename (str).
+
+        Returns:
+            A tuple of transformed images (as a 4-tuple of torch.Tensors), labels (as a list of ints), and filenames (as a list of strs).
+
+        """
+       
+        transforms_0 = [self.gridcreator0.forward(self.transforms[0](item[0])) for item in batch]
+        transforms_1 = [self.gridcreator1.forward(self.transforms[1](item[0])) for item in batch]
```

Comment:
Danger! This is super slow. The reason is that the transformations in `self.transforms` happen before the resize & crop. Image transformations are pretty expensive and we should run them always after resize&crop on the smaller images.